### ISSUE TITLE
https://api.{environment}/billing/billing-entries

### ISSUE THREAD

**Issue by zhangxiang33** [allegro/allegro-api#9082]
Hello, could you please clarify whether the generation logic of the billing entries endpoint "https://api/.{environment}/billing/billing-entries" in Allegro is synchronous with the orders or if there is a delay in receiving the billing information? When we make a request to fetch orders, we also call this endpoint to retrieve commission and VAT-related amounts, but sometimes we receive an empty response. Could you please provide more information on this matter?

**Comment by PrzemyslawLukanowski** [comment]
Hello, the billing entry should be returned almost immediately. You can send me a sample order id and trace id where we returned an empty response, so I will check it out.

**Comment by zhangxiang33** [comment]
8205c600-13cb-11ef-8204-edfbd9f59bc9  This is the order ID. I noticed that the request to the billing entries endpoint does not return a tracking ID. We made requests for this order between May 17th and May 19th, but there was no information returned. However, when I retried it today, I received a response. Could you please investigate why the response was empty during those days?

**Comment by PrzemyslawLukanowski** [comment]
Thank you, I've checked our logs and I can't confirm that the response was empty. For example, trace-id c860aa06631304e2, 
2024-05-17 13:48:27.812, application ***fc0aca7ca, user **27850. Maybe you have been authenticated as another user with no entries? Please let me know when this issue occurs once again.


<br class="Apple-interchange-newline">

**Comment by zhangxiang33** [comment]
The response appears in the following format: "{"billingEntries":[]}", where the "billingEntries" array is empty. It is not returning null as a whole.

**Comment by PrzemyslawLukanowski** [comment]
Yes, I understand, but our logs indicate that there wasn't an empty response for this user, that's why I have the assumption that you were using a token for a different seller. Without a sample trace-id from your side, I can't verify this issue.

**Comment by zhangxiang33** [comment]
Hello, I observed that it still returns "{"billingEntries":[]}" today. This is trace-id: f84affcea31e0214. Could you please help me investigate the cause of this issue?

**Comment by PrzemyslawLukanowski** [comment]
Hello, this operation was performed by a different user than before - xxx458, so I assume on this account ('razxxx70') you don't have any billing entries on [this page ](https://allegro.pl/moje-allegro/moje-konto/rachunki/rozliczenia-z-allegro). Could you please log in and confirm it?

**Comment by zhangxiang33** [comment]
I apologize for the confusion. I misunderstood your statement. If there is no response at this time, it means that the endpoint is not returning any data. This could be because we normally fetch this API after retrieving the orders.

**Comment by zhangxiang33** [comment]
I tested again and after approximately 2 minutes, I fetched the data again and received a response.

**Comment by zhangxiang33** [comment]
c5052af4f270351f   The request was successful this time, as there were two requests made in close proximity. However, I would like to inquire about the specific situation when a null response was received. Can you provide more details about what happened during that particular instance?

**Comment by PrzemyslawLukanowski** [comment]
We shouldn't return null, but if received it, please send us trace-id. In this topic you mentioned that you didn't receive null:

> The response appears in the following format: "{"billingEntries":[]}", where the "billingEntries" array is empty. It is not returning null as a whole.

**Comment by zhangxiang33** [comment]
I apologize for the confusion earlier. The response you received, "{ "billingEntries": [] }" indicates that there are no billing entries associated with the provided trace-id "f84affcea31e0214". This means that there are no billing records or transactions linked to that specific trace-id. Again, I apologize for any inconvenience caused.

**Comment by PrzemyslawLukanowski** [comment]
No worries :)

**Comment by zhangxiang33** [comment]
Could you please provide an explanation

**Comment by zhangxiang33** [comment]
Regarding the question I asked above

**Comment by zhangxiang33** [comment]
> 我们不应该返回 null，但如果收到它，请向我们发送 trace-id。在本主题中，你提到你没有收到 null：
> 
> > 响应按以下格式显示：“{”billingEntries“：[]}”，其中“billingEntries”数组为空。它不会作为一个整体返回 null。

What I meant was not to return null, but to return {"billing Entries": []}. I have already explained this and provided trace id. Could you please help me take a look, I observed that it still returns "{"billingEntries":[]}" today. This is trace-id: f84affcea31e0214.

**Comment by zhangxiang33** [comment]
Sorry, there may be some issues with my description. I found the log today and the request was on 05.06.24. There was no problem because we returned {"billing Entries": []} during our first request, which is why we attempted to retry the request for trace id c5052af4f270351f for the second time. I just want to know why we returned {"billing Entries": []} during our first request

**Comment by PrzemyslawLukanowski** [comment]
Because during this time there were no billing entries for the order provided in the query parameter.

**Comment by zhangxiang33** [comment]
Isn't the billing entry for the order generated together with the order? According to customer feedback, there is a billing entry in the backend when the order is generated. If, as you mentioned, there is no billing entry during this period, could you please ask approximately when the billing entry was generated
